src/validators/travel.rs

```diff
@@ -88,28 +82,28 @@ impl<'a> Handler<'a> for TravelValidator<'a> {
                 }
             }
             TES3Object::Creature(creature) => {
-                if let Some(destinations) = creature.get_destinations() {
-                    if !destinations.is_empty() {
-                        self.caravaners
-                            .insert(creature.id.to_ascii_lowercase(), Caravaner::new(creature));
-                    }
+                let destinations = creature.get_destinations();
+                if !destinations.is_empty() {
+                    self.caravaners
+                        .insert(creature.id.to_ascii_lowercase(), Caravaner::new(creature));
                 }
             }
             TES3Object::Npc(npc) => {
-                if let Some(destinations) = npc.get_destinations() {
-                    if !destinations.is_empty() {
-                        self.caravaners
-                            .insert(npc.id.to_ascii_lowercase(), Caravaner::new(npc));
-                        return;
-                    }
+                let destinations = npc.get_destinations();
+                if !destinations.is_empty() {
+                    self.caravaners
+                        .insert(npc.id.to_ascii_lowercase(), Caravaner::new(npc));
+                    return;
                 }
-                if let Some(class) = &npc.class {
-                    if self.classes.contains(&class.to_ascii_lowercase().as_str()) {
-                        println!(
-                            "Npc {} has class {} but does not offer travel services",
-                            npc.id, class
-                        );
-                    }
+                // TODO: no alloc (use 'uncased' library?)
```

Comment:
I was annoyed by how hard Rust makes it to get heterogeneous lookups in hashmaps using a custom hash and equality implementation. (i.e. I wanted to have https://gitlab.com/OpenMW/openmw/-/blob/master/components/misc/strings/algorithm.hpp )

Of course I ended up removing most of my use cases when I started doing code gen, so that was nice.

Reply:
Oops, forgot I left that comment in. Was a mental note for myself. In theory using a library like `uncased` would enable automatic case-insensitive hashing and comparison. Which might cut out a ton of heap allocations, all the `.to_ascii_lowercase` and such. In practice `mimalloc` is very efficient though so it might not make noticeable difference.

Reply:
Something to consider once we have so many checks we need the speed up 😛